src/libs/ReportUtils.js

```diff
@@ -213,6 +213,18 @@ function isCanceledTaskReport(report = {}, parentReportAction = {}) {
     return false;
 }
 
+/**
+ * Checks if a canceled task should be hidden.
+ * When a canceled task does not have any visible actions, it can be hidden in report screen and LHN.
+ *
+ * @param {Object} report - Represents the parent Chat Report of the canceled task.
+ * @param {Object} parentReportAction - Represents the task preview action of the canceled task in Chat Report
+ * @returns {Boolean}
+ */
+function shouldHideCanceledTaskReport(report = {}, parentReportAction = {}) {
+    return isCanceledTaskReport(report, parentReportAction) && !ReportActionsUtils.isDeletedParentAction(parentReportAction);
```

Comment:
@rojiphil How are we finding if the thread has children or not?

Reply:
@abdulrahuman5196 
The utility function `isDeletedParentAction` internally checks if `childVisibleActionCount` is > 0. That is how it understands that it has children. I think the name `isDeletedParentAction` is misleading. However, the utility function is used in many places and is also in use for a long time. Maybe, the utility function can be renamed whenever a cleanup activity is taken up in the future.

Reply:
https://github.com/Expensify/App/assets/46707890/8ca22c11-a07b-423f-9501-153f077e46b5



@rojiphil I think even the core fix is affected by BE changes for remote devices on chat page also. Correct me if wrong.

Reply:
Am I grasping the core fix correctly? Or am I gauging more fixes not covered as part of core fix?

Reply:
@abdulrahuman5196 
Just tested the remote scenario for the core fix. Here too, the root cause for the problem at the remote device is due to missing pusher updates. Here is an attached test video for the same.

This problem is there even in the test video of the OP. There, the testing team refreshed the page. I am so, wondering if QA considers this an issue at all. May be, this is why BE team also considers this as a pretty minor bug

https://github.com/Expensify/App/assets/3004659/bfb15401-fb90-461f-9079-d3defdd10902